sdk/eventgrid/eventgrid/src/eventGridClient.ts

```diff
@@ -140,10 +163,20 @@ export class EventGridPublisherClient<T extends InputSchema> {
           );
         }
         case "CloudEvent": {
+          // The underlying REST API expects a header named `aeg-channel-name`, and so the generated client
+          // expects that options bag has a property called `aegChannelName`, where as we expose it with the
+          // friendlier name "channelName". Fix up the impedence mismatch here
+          const { channelName, ...sendOptions } = updatedOptions as CloudEventSendOptions;
+
+          if (channelName) {
+            (sendOptions as GeneratedClientPublishCloudEventEventsOptionalParams).aegChannelName =
+              channelName;
+          }
```

Comment:
I can save you one `as` 😄 
```suggestion
          const {
            channelName,
            ...sendOptions
          }: { channelName?: string } & GeneratedClientPublishCloudEventEventsOptionalParams =
            updatedOptions as CloudEventSendOptions;

          if (channelName) {
            sendOptions.aegChannelName = channelName;
          }
```

Reply:
Thanks!  This makes sense to me once you see how the trick is done :-)